### PR TITLE
fix(avoidance): fix previous data inconsistency

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

Close https://github.com/autowarefoundation/autoware.universe/issues/4178?notification_referrer_id=NT_kwDOAqz13LM2OTcyODU5NTcyOjQ0ODg5NTY0

If it fails to generate avoidance path, module doesn't update any previous data.

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/57fbd44b-3f7a-5c19-9b50-2240cddc0319?project_id=prd_jt)

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by satoshi-ota** [comment]
@rej55 I would like you to review and approve as code owner if you're available. @ismetatabay san has already confirmed that this PR resolves https://github.com/autowarefoundation/autoware.universe/issues/4178.

**Review by rej55 — APPROVED** [review]

**Pull request merged**
